core/src/main/java/hudson/Functions.java

```diff
@@ -513,7 +513,15 @@ public static Map getEnvVars() {
     public static boolean isWindows() {
         return File.pathSeparatorChar==';';
     }
-    
+
+    public static boolean isMac() {
+        String osType = System.getProperty("os.name");
+        if (osType == null || osType.isEmpty() || !osType.toLowerCase(Locales.ENGLISH).contains("mac os x")) {
```

Comment:
Can this particular property be null? Should we fail if it is null? or at the very least log this.

Reply:
[Javadoc for System.getProperties](https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#getProperties--) says:

> This set of system properties **always** includes values for the following keys:
> `os.name`
> ...

By that javadoc, it cannot be null.  Null check is harmless and no risk.